### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by craigwblake-openlaw**

### CODE REVIEW

**Comment by craigwblake-openlaw** [comment]
I think this converts most of the Eithers to the new aliased type.  I think we should merge this and test it well before making any more deeper changes.

File: shared/src/main/scala/org/adridadou/openlaw/parser/template/OpenlawTemplateLanguageParserService.scala [hunk 1]
```diff
@@ -8,49 +8,43 @@ import org.adridadou.openlaw.parser.template.printers.XHtmlAgreementPrinter.Frag
...
-      case Left(ex) =>
-        Left(Option(ex.getMessage).getOrElse(""))
-      case Right(result) =>
-        validate(result)
-    }) match {
-      case Success(result) =>
-        result
-      case Failure(ex) =>
-        Left(Option(ex.getMessage).getOrElse(""))
+    compiler.rootRule.run() match {
```
Comment (adridadou) [inline shared/src/main/scala/org/adridadou/openlaw/parser/template/OpenlawTemplateLanguageParserService.scala]: the issue here is that we throw exceptions within the parser. So if you don't use Try here, you may end up with an unhandled exception

Comment (adridadou) [reply shared/src/main/scala/org/adridadou/openlaw/parser/template/OpenlawTemplateLanguageParserService.scala]: maybe use attempt here

Comment (craigwblake-openlaw) [reply shared/src/main/scala/org/adridadou/openlaw/parser/template/OpenlawTemplateLanguageParserService.scala]: Yes, good catch.  Missed this one.

Comment (craigwblake-openlaw) [reply shared/src/main/scala/org/adridadou/openlaw/parser/template/OpenlawTemplateLanguageParserService.scala]: The parser is a particularly odd API in that it returns a Try but also seems to throw exceptions outside of it.

**Pull request merged**